control-plane/csi-driver/src/bin/node/node.rs

```diff
@@ -383,31 +383,49 @@ impl node_server::Node for Node {
             ));
         }
         let _guard = csi_driver::limiter::VolumeOpGuard::new_str(&msg.volume_id)?;
-        match sys::statfs::statfs(&*msg.volume_path) {
-            Ok(info) => Ok(Response::new(NodeGetVolumeStatsResponse {
-                usage: vec![
-                    csi::VolumeUsage {
-                        total: info.blocks() as i64 * info.block_size(),
-                        unit: csi::volume_usage::Unit::Bytes as i32,
-                        available: info.blocks_available() as i64 * info.block_size(),
-                        used: (info.blocks() - info.blocks_free()) as i64 * info.block_size(),
-                    },
-                    csi::VolumeUsage {
-                        total: info.files() as i64,
-                        unit: csi::volume_usage::Unit::Inodes as i32,
-                        available: info.files_free() as i64,
-                        used: (info.files() - info.files_free()) as i64,
+
+        let volume_path = Path::new(&msg.volume_path);
+        if volume_path.exists() {
+            // Check if its a filesystem.
+            if volume_path.is_dir() {
```

Comment:
For block volume the volume path would be a file. Thus the above condition will only be true for the filesystem. This  is in accordance to the logic we use in other methods. We have tested this.

Reply:
Good description, please add this as a code comment itself, so that it will be self-explanatory in the future.